### PR TITLE
LPD-42985 Subscription option on ellipsis menu: D&M

### PR DESCRIPTION
**Pull request by balazssk**
Story link [LPD-42444)](https://liferay.atlassian.net/browse/LPD-42444)
Technical Task link [LPD-42985](https://liferay.atlassian.net/browse/LPD-42985)

As a content creator
I want to be able to subscribe to documents & media
So that I can get notifications of that content.

Functional tests will be included on https://liferay.atlassian.net/browse/LPD-44346

### CODE REVIEW

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/SubscribeFileEntryMVCActionCommand.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+						SessionMessages.KEY_SUFFIX_REFRESH_PORTLET,
+					DLPortletKeys.DOCUMENT_LIBRARY);
+
+				String redirect = _portal.escapeRedirect(
+					ParamUtil.getString(actionRequest, "redirect"));
+
+				if (Validator.isNotNull(redirect)) {
+					sendRedirect(actionRequest, actionResponse, redirect);
+				}
+			}
```
[the remark this answers is not in the document]
Comment (balazssk) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/SubscribeFileEntryMVCActionCommand.java:67]: this is a common pattern from action commands

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/UnsubscribeFileEntryMVCActionCommand.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		long fileEntryId = ParamUtil.getLong(actionRequest, "fileEntryId");
+
+		try {
+			_dlAppService.unsubscribeFileEntry(
+				themeDisplay.getScopeGroupId(), fileEntryId);
+
+			if (SessionErrors.isEmpty(actionRequest)) {
```
[the remark this answers is not in the document]
Comment (balazssk) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/UnsubscribeFileEntryMVCActionCommand.java:54]: this is a common pattern from action commands

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/util/DLSubscriptionUtil.java [hunk 3]
```diff
@@ -23,6 +25,42 @@
...
+		boolean subscribedToAncestor =
+			SubscriptionLocalServiceUtil.isSubscribed(
+				companyId, userId, DLFolder.class.getName(), folderIdsArray);
+
+		boolean subscribed = SubscriptionLocalServiceUtil.isSubscribed(
+			companyId, userId, DLFileEntry.class.getName(), fileEntryId);
+
+		if (subscribed || subscribedToAncestor) {
+			return true;
+		}
```
[the remark this answers is not in the document]
Comment (balazssk) [reply modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/util/DLSubscriptionUtil.java]: updated with https://github.com/liferay-content-management/liferay-portal/pull/6177/commits/c91e748a2b162424efcc07fd0c3dbd42cc553f47

**Review by adolfopa — APPROVED** [review]

**Comment by balazssk** [comment]
Test failures:
1. unrelated: semver error is for Objects
2. unrelated: only DL related files are changed
3. this looks like a commerce Role related failure
4. unrelated: no vulcan changes
5. unrelated: no group changes

**Comment by balazssk** [comment]
found an NPE during test writing, investigating

**Pull request closed without merging**